prog/kube-utils/main.go

```diff
@@ -110,9 +110,19 @@ func reclaimRemovedPeers(weave *weaveapi.Client, cml *configMapAnnotations, node
 		for _, peer := range storedPeerList.Peers {
 			peerMap[peer.NodeName] = peer
 		}
+		// 1a. Check if the list held a peer with my NodeName but different PeerName
+		var deleteOldMe *peerInfo
+		if oldMe, found := peerMap[myNodeName]; found && oldMe.PeerName != myPeerName {
```

Comment:
`storedPeerList` might contain multiple entries with the same `NodeName`. Only one of such entries can be stored in `peerMap` (line 111), so we are risking of not removing other entries.

Reply:
Good spot.  That can happen if the problem has re-occurred over time.